Allow running builds sequentially

vpratz opened the pull request:
In some settings, memory/disk space constraints can make running the builds in parallel not feasible (especially when there are many tags/branches). This PR adds an option to `Driver` to run the builds sequentially. One could also think about surfacing this option to the CLI.

Note that I am not very familiar with `asyncio`, so if you want to incorporate this maybe you want to use a more idiomatic way instead of the loop over gather.

real-yfprojects commented:
Thank you very much for this PR! I did some minor refactoring (see my commit). Would you mind adding a note to docs about this option?

> One could also think about surfacing this option to the CLI.

Sounds like a good idea.

vpratz commented:
Nice, thanks for the refactoring! I will add the docs and take a closer look at the CLI.

vpratz commented:
I have added the docs and slightly modified the existing text. I also added the command line option `--sequential` to set `SEQUENTIAL` from the CLI. Please modify as you see fit.

vpratz commented:
I just noticed that I overlooked updating the README, I will add those changes soon.
